float8_experimental/float8_linear.py

```diff
@@ -154,8 +160,9 @@ def __init__(self, *args, **kwargs):
         )
         self.register_always_float32_buffer("fp8_scale_dL_dY", torch.tensor([1.0]))
 
-        # Whether to emulate the fp8 matmul logic in float32
-        self.emulate = False
+        # Defines the behavior of the matmul in the forward and backward pass
+        self.forward_config = ScaledMMConfig()
+        self.backward_config = ScaledMMConfig()
```

Comment:
is it possible to configure the two backward gemms separately?

Reply:
This is somewhat challenging , since as written today we dont have a very clean way of knowing which matmul is which

cc @bdhirsh maybe Im no thinking of something.

We have `out = x@W` where x = FLoat8Tensor and W = Float8Tensor. 

Since W will not be used in calucalting the gradW you could tag some extra info on the activation float8tensor and since this gets used for backward this should get carried through to backwards calcs.

I think that this would be better as a follow up though since the logic gets spread out over multiple Float8Tensor instances.

Reply:
> since as written today we dont have a very clean way of knowing which matmul is which

yeah, this is weird because the config is really per gemm but we have to stick it on a tensor.  How about something like
1. local: given matmul(A, B), the B matmul (second argument) always overrides A (first argument).
2. global: the float8 UX allows setting options for the 3 gemms, and under the hood maps it to be implemented via (1).

While not the most intuitive to implement, I think that could work?

Reply:
separate PR sgtm, I do feel like we need to make all 3 gemms configurable before we lock the API down.